flexmeasures/ui/tests/test_asset_crud.py

```diff
@@ -134,6 +140,37 @@ def test_edit_asset(db, client, setup_assets, requests_mock, as_admin):
     assert str(mock_asset["longitude"]) in str(response.data)
 
 
+def test_sensors_to_show_as_kpis_json(
+    db, client, setup_assets, requests_mock, as_admin
+):
+    mock_asset = mock_asset_response_with_kpis(db=db, as_list=False)
+    requests_mock.patch(f"{api_path_assets}/1", status_code=200, json=mock_asset)
+    requests_mock.get(f"{api_path_assets}/1", status_code=200, json=mock_asset)
+
+    # Test asset with invalid json
+    ma_copy = copy.deepcopy(mock_asset)
+    ma_copy["sensors_to_show_as_kpis"] = "not a json."
+    response = client.post(
+        url_for("AssetCrudUI:post", id=1),
+        follow_redirects=True,
+        data=mock_api_data_as_form_input(ma_copy),
+    )
+    assert response.status_code == 200
+    assert b"Cannot edit asset:" in response.data
```

Comment:
I recommend checking that the response explicitly mentions an expected reason for not being able to edit the asset. Likewise in line 171.